src/com/google/enterprise/adaptor/filenet/Params.java

```diff
@@ -0,0 +1,197 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.enterprise.adaptor.filenet;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableSet;
+import com.google.enterprise.adaptor.Config;
+import com.google.enterprise.adaptor.InvalidConfigurationException;
+
+import java.net.URISyntaxException;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+// TODO(bmj): move this into FileNetAdaptor?
+class Params {
+  private static final Logger logger =
+      Logger.getLogger(Params.class.getName());
+
+  private static final Splitter splitter =
+      Splitter.on(',').omitEmptyStrings().trimResults();
+
+  private final String contentEngineUrl;
+  private final String username;
+  private final String password;
+  private final String objectStore;
+
+  private final String objectFactory;
+  private final String displayUrl;
+  private final boolean isPublic;
+  private final boolean pushAcls;
+  private final String additionalWhereClause;
+  private final String deleteAdditionalWhereClause;
+  private final Set<String> includedMetadata;
+  private final Set<String> excludedMetadata;
+  private final String globalNamespace;
+  private final int maxFeedUrls;
+
+  public Params(Config config) throws InvalidConfigurationException {
+    contentEngineUrl = config.getValue("filenet.contentEngineUrl");
+    logger.log(Level.CONFIG, "filenet.contentEngineUrl: {0}", contentEngineUrl);
+    try {
+      new ValidatedUri(contentEngineUrl).logUnreachableHost();
+    } catch (URISyntaxException e) {
+      throw new InvalidConfigurationException(
+          "Invalid filenet.contentEngineUrl: " + e.getMessage());
+    }
+
+    objectStore = config.getValue("filenet.objectStore").trim();
```

Comment:
Trim is redundant, the library trims all config values since 4.0.4 or something like that.